inference_overview.md

```diff
@@ -0,0 +1,450 @@
+# B&A Inference Explainer
```

Comment:
Can we also update README to include this doc link?

Reply:
Will update README in a separate commit, since it references Protected Audience.